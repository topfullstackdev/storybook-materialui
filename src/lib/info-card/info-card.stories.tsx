import Box from '@mui/material/Box';
import { Meta, Story } from '@storybook/react';

import { Color, InfoCard, InfoCardProps } from './info-card';

export default {
  component: InfoCard,
  title: 'InfoCard',
  argTypes: {
    children: {
      table: {
        disable: true,
      },
    },
    color: {
      name: 'color',
      options: Color,
      control: { type: 'select' },
    },
  },
} as Meta;

const Template: Story<InfoCardProps> = function Template(args) {
  const { children, ...rest } = args;
  return (
    <InfoCard {...rest}>
      <Box component="span" sx={{ fontSize: 12 }}>
        Advanced Marketing
      </Box>
      <Box component="span" sx={{ fontSize: 12 }}>
        Advanced Sales
      </Box>
      <Box component="span" sx={{ fontSize: 12 }}>
        Appraisals
      </Box>
    </InfoCard>
  );
};

export const Primary = Template.bind({});
Primary.args = {
  statLabel: 'Skills Developing',
  statNumber: 88,
};
