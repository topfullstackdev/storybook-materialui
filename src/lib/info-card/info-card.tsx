import styled from '@emotion/styled';
import Box from '@mui/material/Box';

export enum Color {
  PURPLE = 'rgb(128, 55, 155)',
  ORANGE = 'rgb(232, 119, 34)',
  TEAL = 'rgb(0, 165, 153)',
  BLUE = 'rgb(0, 161, 222)',
}

export interface InfoCardProps {
  color: Color;
  statNumber: number;
  statLabel: string;
  children?: React.ReactNode;
}

type StyledBoxProps = {
  color: Color;
};

const StyledBox = styled(Box)<StyledBoxProps>`
  border: 1px solid ${({ color }) => color};
  width: 100%;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  color: ${({ color }) => color};
`;

export function InfoCard({
  color,
  statNumber,
  statLabel,
  children,
}: InfoCardProps) {
  return (
    <Box sx={{ width: '100%' }}>
      <StyledBox color={color}>
        <Box
          data-cy={`Count:${statNumber}`}
          component="span"
          sx={{ fontWeight: 700, fontSize: 32, textAlign: 'center' }}
        >
          {statNumber}
        </Box>
        <Box
          data-cy={`Label:${statLabel}`}
          component="span"
          sx={{ fontSize: 14, textAlign: 'center' }}
        >
          {statLabel}
        </Box>
      </StyledBox>
      {children && <StyledBox color={color}>{children}</StyledBox>}
    </Box>
  );
}
