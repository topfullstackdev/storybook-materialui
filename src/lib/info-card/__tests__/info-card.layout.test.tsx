import Box from '@mui/material/Box';
import { render } from '@testing-library/react';

import { Color, InfoCard } from '../info-card';

describe('Text', () => {
  it('should render successfully', () => {
    const { container } = render(
      <InfoCard
        color={Color.BLUE}
        statNumber={31}
        statLabel="Skills Developing"
      >
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Marketing
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Sales
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Appraisals
        </Box>
      </InfoCard>,
    );
    expect(container).toMatchSnapshot();
  });

  it('statNumber should be in the component, fontsize should be 32px', () => {
    const { getByText, container } = render(
      <InfoCard
        color={Color.BLUE}
        statNumber={31}
        statLabel="Skills Developing"
      >
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Marketing
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Sales
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Appraisals
        </Box>
      </InfoCard>,
    );
    const statNumber = getByText('31');
    const style = window.getComputedStyle(statNumber);
    expect(container).toMatchSnapshot();
    expect(style.fontSize).toEqual('32px');
  });

  it('statLabel should be in the component, fontsize should be 14px', () => {
    const { getByText, container } = render(
      <InfoCard
        color={Color.BLUE}
        statNumber={31}
        statLabel="Skills Developing"
      >
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Marketing
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Sales
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Appraisals
        </Box>
      </InfoCard>,
    );
    const statLabel = getByText('Skills Developing');
    const style = window.getComputedStyle(statLabel);
    expect(container).toMatchSnapshot();
    expect(style.fontSize).toEqual('14px');
  });

  it('children should be display in the component when children is existing', () => {
    const { container } = render(
      <InfoCard
        color={Color.BLUE}
        statNumber={31}
        statLabel="Skills Developing"
      >
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Marketing
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Sales
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Appraisals
        </Box>
      </InfoCard>,
    );
    expect(container).toMatchSnapshot();
  });

  it('border color should be BLUE  when color is Color.BLUE', () => {
    const { getByText, container } = render(
      <InfoCard
        color={Color.BLUE}
        statNumber={31}
        statLabel="Skills Developing"
      >
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Marketing
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Advanced Sales
        </Box>
        <Box component="span" sx={{ fontSize: 12 }}>
          Appraisals
        </Box>
      </InfoCard>,
    );
    expect(container).toMatchSnapshot();
    const element = getByText('Skills Developing').closest('div');
    const style = element && window.getComputedStyle(element);
    expect(style?.borderColor).toEqual(Color.BLUE);
  });
});
