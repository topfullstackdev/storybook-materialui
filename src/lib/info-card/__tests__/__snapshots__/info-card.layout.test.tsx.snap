// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text border color should be BLUE  when color is Color.BLUE 1`] = `
.emotion-0 {
  width: 100%;
}

.emotion-2 {
  color: rgb(0, 161, 222);
  border: 1px solid rgb(0, 161, 222);
  width: 100%;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(0, 161, 222);
}

.emotion-3 {
  font-weight: 700;
  font-size: 32px;
  text-align: center;
}

.emotion-4 {
  font-size: 14px;
  text-align: center;
}

.emotion-7 {
  font-size: 12px;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-3"
        data-cy="Count:31"
      >
        31
      </span>
      <span
        class="MuiBox-root emotion-4"
        data-cy="Label:Skills Developing"
      >
        Skills Developing
      </span>
    </div>
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Marketing
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Sales
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Appraisals
      </span>
    </div>
  </div>
</div>
`;

exports[`Text children should be display in the component when children is existing 1`] = `
.emotion-0 {
  width: 100%;
}

.emotion-2 {
  color: rgb(0, 161, 222);
  border: 1px solid rgb(0, 161, 222);
  width: 100%;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(0, 161, 222);
}

.emotion-3 {
  font-weight: 700;
  font-size: 32px;
  text-align: center;
}

.emotion-4 {
  font-size: 14px;
  text-align: center;
}

.emotion-7 {
  font-size: 12px;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-3"
        data-cy="Count:31"
      >
        31
      </span>
      <span
        class="MuiBox-root emotion-4"
        data-cy="Label:Skills Developing"
      >
        Skills Developing
      </span>
    </div>
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Marketing
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Sales
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Appraisals
      </span>
    </div>
  </div>
</div>
`;

exports[`Text should render successfully 1`] = `
.emotion-0 {
  width: 100%;
}

.emotion-2 {
  color: rgb(0, 161, 222);
  border: 1px solid rgb(0, 161, 222);
  width: 100%;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(0, 161, 222);
}

.emotion-3 {
  font-weight: 700;
  font-size: 32px;
  text-align: center;
}

.emotion-4 {
  font-size: 14px;
  text-align: center;
}

.emotion-7 {
  font-size: 12px;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-3"
        data-cy="Count:31"
      >
        31
      </span>
      <span
        class="MuiBox-root emotion-4"
        data-cy="Label:Skills Developing"
      >
        Skills Developing
      </span>
    </div>
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Marketing
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Sales
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Appraisals
      </span>
    </div>
  </div>
</div>
`;

exports[`Text statLabel should be in the component, fontsize should be 14px 1`] = `
.emotion-0 {
  width: 100%;
}

.emotion-2 {
  color: rgb(0, 161, 222);
  border: 1px solid rgb(0, 161, 222);
  width: 100%;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(0, 161, 222);
}

.emotion-3 {
  font-weight: 700;
  font-size: 32px;
  text-align: center;
}

.emotion-4 {
  font-size: 14px;
  text-align: center;
}

.emotion-7 {
  font-size: 12px;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-3"
        data-cy="Count:31"
      >
        31
      </span>
      <span
        class="MuiBox-root emotion-4"
        data-cy="Label:Skills Developing"
      >
        Skills Developing
      </span>
    </div>
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Marketing
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Sales
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Appraisals
      </span>
    </div>
  </div>
</div>
`;

exports[`Text statNumber should be in the component, fontsize should be 32px 1`] = `
.emotion-0 {
  width: 100%;
}

.emotion-2 {
  color: rgb(0, 161, 222);
  border: 1px solid rgb(0, 161, 222);
  width: 100%;
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgb(0, 161, 222);
}

.emotion-3 {
  font-weight: 700;
  font-size: 32px;
  text-align: center;
}

.emotion-4 {
  font-size: 14px;
  text-align: center;
}

.emotion-7 {
  font-size: 12px;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-3"
        data-cy="Count:31"
      >
        31
      </span>
      <span
        class="MuiBox-root emotion-4"
        data-cy="Label:Skills Developing"
      >
        Skills Developing
      </span>
    </div>
    <div
      class="emotion-1 MuiBox-root emotion-2"
    >
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Marketing
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Advanced Sales
      </span>
      <span
        class="MuiBox-root emotion-7"
      >
        Appraisals
      </span>
    </div>
  </div>
</div>
`;
